app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -100,14 +127,29 @@ class MainActivity : AppCompatActivity() {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults)
         if(requestCode == 100){ //result requestCode to match with a requestCode given in requestPermissions(), this can be any >0 value
             if( grantResults.size == 2 && grantResults[0] == PackageManager.PERMISSION_GRANTED && grantResults[1] == PackageManager.PERMISSION_GRANTED){
-                enableButton()
+                enableGetWeatherButton()
             } else {
                 Toast.makeText(this@MainActivity, "App requires location permission to work properly!",Toast.LENGTH_LONG).show()
                 checkPermissions()
             }
         }
     }
 
+
+    private fun disableButton(button: Button, resid: Int, residToast: Int) {
+        button.setOnClickListener {
+            Toast.makeText(this@MainActivity, resources.getText(residToast), Toast.LENGTH_LONG).show()
+        }
+        button.setBackgroundColor(Color.GRAY)
+        button.setText(resid)
+    }
+
+    private fun enableButton(button: Button, resid: Int) {
+        button.isClickable = true
```

Comment:
Same here, to enable just set isEnabled to true.

Reply:
same, changed to isEnabled true